local/compose/pull.go

```diff
@@ -64,73 +67,89 @@ func (s *composeService) Pull(ctx context.Context, project *types.Project) error
 			continue
 		}
 		eg.Go(func() error {
-			w.Event(progress.Event{
-				ID:     service.Name,
-				Status: progress.Working,
-				Text:   "Pulling",
-			})
-			ref, err := reference.ParseNormalizedNamed(service.Image)
+			err := s.pullServiceImage(ctx, service, info, configFile, w)
 			if err != nil {
-				return err
+				if !opts.IgnoreFailures {
+					return err
+				}
+				w.Event(progress.Event{
```

Comment:
event should be created before returning error, so user see it reported on progress before the command fails

Reply:
This is done inside `err := s.pullServiceImage(ctx, service, info, configFile, w)` just some lines before. It calls https://github.com/docker/compose-cli/pull/1454/files#diff-49c56ffdcd4d1b6c21e27379db32f411e24665476483f4901e99359d2eb83177R155

Reply:
So if all event display is done inside `pullServiceImage` (that makes sense), we should not have this event displayed here, just check the ignoreFailures option and return the err or nil ?

Reply:
OK, I understand now, this is a workaround to still display the error message when discarding the error with --ignore-failure. And symetrical to what we do with push